File: Data Structures & Algorithms/validate-parentheses/submission-4.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for ch in s:
            if ch == "[" or ch == "(" or ch == "{":
                stack.append(ch)
            else:
                top = stack.pop()
                if ch == "]" and top == "[":
                    continue
                elif ch == ")" and top == "(":
                    continue
                elif ch == "}" and top == "{":
                    continue
                else:
                    return False

        return True

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.js
class Solution {
    /**
     * @param {string[]} tokens
     * @return {number}
     */
    evalRPN(tokens) {
        let stack = [];
        for (let token of tokens) {
            if (token == "+" || token == "-" || token == "*" || token == "/") {
                const pop1 = stack.pop();
                const pop2 = stack.pop();

                if (token == "+") {
                    stack.push(pop2 + pop1);
                } else if (token == "-") {
                    stack.push(pop2 - pop1);
                } else if (token == "*") {
                    stack.push(pop2 * pop1);
                } else stack.push(parseInt(pop2 / pop1));
            } else stack.push(parseInt(token));
        }
        return stack.pop();
    }
}
